R/sidebar.R

```diff
@@ -96,14 +96,26 @@ layout_sidebar <- function(
   fillable = FALSE,
   fill = TRUE,
   bg = NULL,
-  border = TRUE,
-  border_radius = TRUE,
+  border = NULL,
+  border_radius = NULL,
   height = NULL
 ) {
   if (!inherits(sidebar, "sidebar")) {
     abort("`sidebar` argument must contain a `bslib::sidebar()` component.")
   }
 
+  if (!(is.null(border) || rlang::is_bare_logical(border, n = 1))) {
+    abort("`border` must be `NULL`, `TRUE`, or `FALSE`.")
+  } else if (!is.null(border) && is.na(border)) {
+    border <- NULL
+  }
+
+  if (!(is.null(border_radius) || rlang::is_bare_logical(border_radius, n = 1))) {
+    abort("`border_radius` must be `NULL`, `TRUE`, or `FALSE`.")
+  } else if (!is.null(border_radius) && is.na(border_radius)) {
+    border_radius <- NULL
+  }
```

Comment:
This could certainly be cleaner, but I'd rather return to this with rlang's type checking functions or {checkmate} later than to over optimize these checks.